### PR TITLE
Split log setup from lm_create_ctx() to lm_setup_log()

### PR DESCRIPTION
**Pull request by swapnili**
Signed-off-by: Swapnil Ingle <swapnil.ingle@nutanix.com>

### CODE REVIEW

File: lib/muser_ctx.c [hunk 1]
```diff
@@ -1226,7 +1226,7 @@ lm_ctx_try_attach(lm_ctx_t *lm_ctx)
 }
 
 lm_ctx_t *lm_create_ctx(lm_trans_t trans, const char *path, int flags,
```
Comment (jlevon) [inline lib/muser_ctx.c:1228]: No need to fix now, but we should be starting functions at the start of the line, like:

lm_ctx_t *
lm_create_ctx...

File: lib/muser_ctx.c [hunk 2]
```diff
@@ -1245,8 +1245,6 @@ lm_ctx_t *lm_create_ctx(lm_trans_t trans, const char *path, int flags,
     //FIXME: Validate arguments.
     // Set other context data.
     lm_ctx->pvt = pvt;
-    lm_ctx->log = log;
-    lm_ctx->log_lvl = log_lvl;
```
Comment (jlevon) [inline lib/muser_ctx.c:1249]: Please explicitly init log_lvl still

File: samples/gpio-pci-idio-16.c [hunk 3]
```diff
@@ -109,6 +108,11 @@ main(int argc, char *argv[])
         err(EXIT_FAILURE, "failed to initialize device emulation");
     }
 
+    ret = lm_setup_log(lm_ctx, verbose ? _log : NULL, LM_DBG);
```
Comment (jlevon) [inline samples/gpio-pci-idio-16.c:111]: I think this should be:

lm_setup_log(lm_ctx, _log, verbose ? LM_DBG : LM_ERR);

?

File: samples/server.c [hunk 4]
```diff
@@ -423,11 +423,16 @@ int main(int argc, char *argv[])
     }
 
     server_data.lm_ctx = lm_ctx = lm_create_ctx(LM_TRANS_SOCK, argv[optind], 0,
-            verbose ? _log : NULL, LM_DBG, &server_data);
+                                                &server_data);
     if (lm_ctx == NULL) {
         err(EXIT_FAILURE, "failed to initialize device emulation\n");
     }
 
+    ret = lm_setup_log(lm_ctx, verbose ? _log : NULL, LM_DBG);
```
Comment (jlevon) [inline samples/server.c:431]: ditto

**Review by jlevon — CHANGES_REQUESTED** [review]

File: lib/muser_ctx.c (hunk 1 above)
Comment (swapnili) [reply lib/muser_ctx.c:1228]: Sure

File: lib/muser_ctx.c (hunk 2 above)
Comment (swapnili) [reply lib/muser_ctx.c:1249]: Sure will init to LM_ERR

File: samples/gpio-pci-idio-16.c (hunk 3 above)
Comment (swapnili) [reply samples/gpio-pci-idio-16.c]: Ack

File: samples/server.c (hunk 4 above)
Comment (swapnili) [reply samples/server.c]: Ack

**Review by jlevon — APPROVED** [review]
thanks!

**Pull request merged**